### PR TITLE
Reduce travis file complexity

### PR DESCRIPTION
**Pull request by cdrini**
We hit Travis' new restrictions for public repos (see https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing ) last month, so trying to see if we can reduce how much time we spend running Travis. We hit the limit pretty late in the month though, on Nov 24th; so a few reductions might be all we need to do to get back into December.

Here I:
1) Only test on Xenial; that's what all our Dockerfiles use, and it's very unlikely we'll make a change that causes something to break on other versions of Ubuntu
2) Don't test on Python 3.9 ; it's very unlikely we'll introduce a change that causes 3.9 to break

That takes us from 6 jobs to 4 jobs :)

Python 3.9 was also the slowest (see e.g. https://travis-ci.com/github/internetarchive/openlibrary/builds/203778599 ), so this should reduce our total travis time by ~7.5min per build -- that's 42% !

### Testing
Need to wait for Dec 1st to test this with Travis

### Stakeholders
@cclauss

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,22 +37,11 @@ jobs:
         - make components
         - npm run test
 
-  allow_failures:
-    - name: Python 2.7 on Infogami master
-
 install:
   - pip install --upgrade pip setuptools wheel
   - pip install --use-feature=2020-resolver -r requirements_test.txt
```
Comment (cclauss) [inline .travis.yml:42]:
```suggestion
  - pip install -r requirements_test.txt
```
Cherrypick #4187

**Comment by cdrini** [comment]
Oooooof, I was entirely wrong; the credits don't review monthly :( The 10,000 credits were a one-time free trial :/

**Comment by cclauss** [comment]
Agreed...  These announcements are far away from Travis CI's previous commitments of _always free for open source projects_.

**Review by mekarpeles — APPROVED** [review]
lgtm

**Pull request merged**